mage_ai/api/resources/PipelineResource.py

```diff
@@ -146,16 +147,33 @@ def update_schedule_status(status, pipeline_uuid):
             for schedule in schedules:
                 schedule.update(status=status)
 
+        @safe_db_query
+        def update_pipeline_run_status(status, pipeline_uuid):
+            pipeline_runs = (
+                PipelineRun.
+                query.
+                filter(PipelineRun.pipeline_uuid == pipeline_uuid).
+                filter(or_(
+                    PipelineRun.status == PipelineRun.PipelineRunStatus.INITIAL,
+                    PipelineRun.status == PipelineRun.PipelineRunStatus.RUNNING,
+                ))
+            )
+            for pipeline_run in pipeline_runs:
```

Comment:
* We can batch update the pipeline run status.
* We also need to cancel the block runs
* I think this method is only used for cancelling pipeline run. So you can just call it `cancel_pipeline_runs`

Reply:
@wangxiaoyou1993 There was a PipelineScheduler method that canceled the pipeline and block runs. Does this work?